aten/src/ATen/native/TensorConversions.cpp

```diff
@@ -72,6 +72,17 @@ Tensor autocast_to_fp16(const Tensor& self) {
   }
 }
 
+// If input tensor is fp32, cast it to fp16, otherwise leave it alone.
+// (this is intended to be used internally by the JIT autocast implementation)
+Tensor autocast_to_bf16(const Tensor& self) {
```

Comment:
Do we support auto casting in AMP for bf16?

Reply:
in upstream master we do. I made a PR to get it into upstream a while back along with some follow up fixes 
https://github.com/pytorch/pytorch/pull/61002
https://github.com/pytorch/pytorch/pull/63798
https://github.com/pytorch/pytorch/pull/64122